walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java

```diff
@@ -11,17 +11,29 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
+ 
 package com.google.sps;
-
+ 
 /**
  * Utility class for creating greeting messages.
  */
 public class Greeter {
   /**
    * Returns a greeting for the given name.
    */
+  //default trimmer 
   public String greet(String name) {
+    return "Hello " + name.trim().replaceAll("[@#$%!? ]","");
+  }
+
+  //custom trimmer
+  private static final String[] invalidArray = {" ","@","%","!"};
```

Comment:
nit: there appears to be some spacing / formatting issues in this file. I suspect running clang should fix it.